src/nodes/snat_node.c

```diff
@@ -78,22 +83,80 @@ static __rte_always_inline rte_edge_t get_next_index(__rte_unused struct rte_nod
 			/* Expect the new destination in this conntrack object */
 			cntrack->flow_flags |= DP_FLOW_FLAG_SRC_NAT;
 			dp_delete_flow(&cntrack->flow_key[DP_FLOW_DIR_REPLY]);
-			cntrack->flow_key[DP_FLOW_DIR_REPLY].ip_dst = ntohl(ipv4_hdr->src_addr);
-			if (snat_data->nat_ip != 0)
+			cntrack->flow_key[DP_FLOW_DIR_REPLY].l3_dst.ip4 = ntohl(ipv4_hdr->src_addr);
+			if (snat_data_p->nat_ip != 0)
 				cntrack->flow_key[DP_FLOW_DIR_REPLY].port_dst = df->nat_port;
 
 			if (DP_FAILED(dp_add_flow(&cntrack->flow_key[DP_FLOW_DIR_REPLY], cntrack)))
 				return SNAT_NEXT_DROP;
 		}
-		// it is deleted due the case where an init packet/request is sent twice, and no action is done if it is returned here
-		// it has to continue to the follow-up code
-		// return 1;
+
+		if (df->l3_type == RTE_ETHER_TYPE_IPV6 && port->iface.nat_ip && dp_is_ip6_in_nat64_range(df->dst.dst_addr6)
+		    && df->flow_type == DP_FLOW_SOUTH_NORTH) {
+			struct snat_data snat_data = {0};
```

Comment:
I thought this was a big no-no for me to do, otherwise I would use this quite often I think. If renamed to `snat64_data` as suggested above, I don't think it needs to be here to indicate that it's a "different snat data"

Reply:
Your are right. I also ask myself, why I declared it here and not in the beginning of the function and the above name change to _p was also to avoid the shadowing. So fixed both.